src/spec/datatypes.rs

```diff
@@ -190,21 +194,87 @@ impl fmt::Display for PrimitiveType {
 }
 
 /// DataType for a specific struct
-#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+#[derive(Debug, Serialize, PartialEq, Eq, Clone)]
 #[serde(rename = "struct", tag = "type")]
 pub struct StructType {
     /// Struct fields
     fields: Vec<StructField>,
+    /// Lookup for index by field id
+    #[serde(skip_serializing)]
```

Comment:
*I have to say this is why I want an in-memory represents :cry:. Since we have already reached a consensus in the previous discussion, I will not initiate a new round here.*

Reply:
I don't want to force this serialization/deserialization design. If we don't have a consensus on the design, I think we should get back to the discussion. It doesn't make sense to merge something that might get changed later on.

Reply:
> if we want to avoid such annotation, we could implement a serializer for `StructType`.

Currently, it appears that there is no immediate need for this. Let's proceed and observe how the situation develops.